src/test/java/emissary/config/ServiceConfigGuideTest.java

```diff
@@ -645,4 +648,52 @@ void testFindRequiredStringEntry() {
         assertNotNull(ex.getMessage(), "Exception message should not be null");
         assertTrue(ex.getMessage().contains(missingProp), "Exception message should name the missing config entry");
     }
+
+    @Test
+    void testFindStringMatchMultiMapOrdered() {
+        // this test fails if ORDERED is set to false due to results ordering
+        final boolean ORDERED = true;
+
+        final ServiceConfigGuide config = new ServiceConfigGuide();
+
+        final String prefix = "MM_";
+        final String key1 = "key1";
+        final String key2 = "key2";
+        final String key3 = "key3";
+        final String value1 = "_value1";
+        final String value2 = "_value2";
+        final String value3 = "_value3";
+        List<String> suffixes = List.of(value1, value2, value3);
+        List<String> keys = List.of(key1, key2, key3);
+
+
+        Map<String, Set<String>> expected = new LinkedHashMap<>();
+        for (String k : keys) {
+            for (String s : suffixes) {
+                String key = prefix + k;
+                String value = k + s;
+                // add prefixed entry to config
+                config.addEntry(key, value);
+                // add non-prefixed entry to expected map (prefix gets stripped during findStringMatchMultiMap call)
+                // UPPERCASE the key to match findStringMatchMultiMap retrieval behavior
+                expected.computeIfAbsent(k.toUpperCase(Locale.getDefault()), v -> new LinkedHashSet<>()).add(value);
+            }
+        }
+        Map<String, Set<String>> actual = config.findStringMatchMultiMap(prefix, ORDERED);
+        assertEquals(expected.size(), actual.size(), "should have the same number of keys");
+        var actualsIterator = actual.entrySet().iterator();
+        var expectedIterator = expected.entrySet().iterator();
+        int i = 0;
+        while (expectedIterator.hasNext()) {
+            assertTrue(actualsIterator.hasNext());
+
+            i++;
+            var expectedEntry = expectedIterator.next();
+            var actualEntry = actualsIterator.next();
+            assertEquals(expectedEntry.getKey(), actualEntry.getKey(), "failed on key #" + i);
+            assertEquals(expectedEntry.getValue(), actualEntry.getValue(), "value mismatch for key '" + actualEntry.getKey() + "'");
+        }
+
```

Comment:
```suggestion
assertTrue(i>0);
```
I know it's pedantic but I've done this myself where I accidentally forgot to fix a test I was temporarily messing with and turns out the tests still pass because the test case list was empty and it never hit the asserts.

Reply:
Addressing this by adding an earlier `assertFalse(actual.isEmpty())`, but that's a good catch